addons/mail/static/src/js/thread.js

```diff
@@ -110,6 +120,12 @@ var Thread = Widget.extend({
             ORDER: ORDER,
         }));
 
+        var attachments = _.flatten(_.map(messages, function(msg){
+            return msg.attachment_ids
+        }));
```

Comment:
No, there is list of thread(message) and that messages have list of attachments.
so that we have fetch all attachments to the all messages and create a array for next and previous button(like: Bootstrap Carousel Plugin)